diesel/src/pg/connection/mod.rs

```diff
@@ -547,6 +547,29 @@ impl PgConnection {
             .set_notice_processor(noop_notice_processor);
         Ok(())
     }
+
+    /// See Postgres documentation for SQL commands Notify and Listen
+    /// The returned iterator can yield items even after a None value when new notifications have been received.
+    pub fn notifications_iter(&self) -> NotificationsIterator<'_> {
+        NotificationsIterator {
+            conn: &self.connection_and_transaction_manager.raw_connection,
+        }
+    }
+}
+
+#[allow(missing_debug_implementations)]
+pub struct NotificationsIterator<'a> {
+    conn: &'a RawConnection,
+}
+
+pub use raw::PGNotification;
+
+impl Iterator for NotificationsIterator<'_> {
+    type Item = PGNotification;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.conn.pqnotifies()
+    }
 }
```

Comment:
I would prefer moving all notifications related code into a separate submodule to keep the code clean

Reply:
With that I'd appreciate your help or better you doing it, please. It's much faster for you to just move the code where you want it. It's hard for me to balance everything what goes into API design and to predict your appreciation of the different factors.

Reply:
I will take care of that in the next days, no worries